src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -47,6 +47,19 @@ internal abstract class AbstractRecommendationServiceBasedCompletionProvider<TSy
                 // Just filter valid symbols. Nothing to preselect
                 return recommendedSymbols.NamedSymbols.SelectAsArray(IsValidForGenericConstraintContext, s => (s, preselect: false));
             }
+            // When inheriting from a class in C# both classes and interfaces are valid completions,
+            // so we need to check it all in one go
+            else if (context.IsInheritanceRequiringClassContext || context.IsInheritanceRequiringInterfaceContext)
+            {
+                ISymbol? inheritingFrom = null;
+
+                // We know for sure that in both C# and VB syntax, that represents a type,
+                // is 2 steps up relatively to our current position
+                if (context.TargetToken.Parent?.Parent is not null and var typeSyntax)
```

Comment:
I'm not getting the parent.Parent check. What if it's partially written. Like you're after the dot Ina qualified name. Tests please